node-backend/app/test/badges.test.js

```diff
@@ -1,19 +1,62 @@
+let path = require("path");
+let dotEnvPath = path.resolve("../.env");
+require("dotenv").config({ path: dotEnvPath });
+const db = require("./dbConnection");
+const data = require("./data");
+const { createToken } = require("./jwtTokenGenration");
+
 const supertest = require("supertest"); //eslint-disable-line node/no-unpublished-require
 const should = require("should" /*eslint-disable-line node/no-unpublished-require*/); //eslint-disable-line no-unused-vars
 
-// This agent refers to PORT where program is runninng.
-
-const server = supertest.agent(process.env.URL);
-const token = process.env.TOKEN;
+const server = supertest.agent(process.env.TEST_URL + process.env.HTTP_PORT);
+let token;
+let id;
+let orgId;
 
 // UNIT test begin
 
-describe(/*eslint-disable-line no-undef*/ "SAMPLE unit test", function () {
+describe(/*eslint-disable-line no-undef*/ "test cases for badges", function () {
+  /*eslint-disable-line no-undef*/ before((done) => {
+    db.organizations.create(data.organizations).then((data) => {
+      orgId = data.id;
+      token = createToken(2, 3, 1);
```

Comment:
what does 2 , 3 , 1 indicates?

Reply:
Also you can write a function where you can pass Role name and can fetch role id from that. Org id and user id you should not hardcode